yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -878,7 +878,21 @@ export class Archiver extends (EventEmitter as new () => ArchiverEmitter) implem
       const publishedCheckpoints = await Promise.all(retrievedCheckpoints.map(b => retrievedToPublishedCheckpoint(b)));
       const validCheckpoints: PublishedCheckpoint[] = [];
 
-      for (const published of publishedCheckpoints) {
+      for (let i = 0; i < retrievedCheckpoints.length; i++) {
+        const retrieved = retrievedCheckpoints[i];
+        const { published, error: publishedCheckpointError } = publishedCheckpoints[i];
+        if (!published) {
+          this.log.warn(
+            `Skipping checkpoint ${retrieved.checkpointNumber} due to inability to convert it to a published checkpoint`,
+            {
+              l1BlockNumber: retrieved.l1.blockNumber,
+              reason: publishedCheckpointError,
+            },
+          );
+
+          continue;
```

Comment:
IIUC we used to throw if there was an error converting the checkpoint object, which halted the archiver sync, whereas we are now continuing, right? Why the change? Not saying we shouldn't, just wondering about the rationale for it since it's a big change. Also, do we know what could trigger an error in the conversion here (aside from a bug)?

Reply:
We purposefully skip the block if attestations are invalid since we know that the next proposer will "invalidate" this block, removing it from the chain. And we skip the block if the archive root does not match the one in rollup storage because it means we're trying to sync a block that has been pruned. In both situations the block we're skipping is not part of the chain.

That's why I'm worried about skipping in other situations. If we have a bug and there's a _valid_ block we can't sync, then we should not skip it, since the following block would fail due to the gap in the chain. So I think I'd go back to failing loudly. WDYT?